ui/raidboss/data/06-ew/trial/endsinger.ts

```diff
@@ -0,0 +1,231 @@
+  zoneId: ZoneId.TheFinalDay,
+  timelineFile: 'endsinger.txt',
+  triggers: [
+    {
+      id: 'Endsinger Doomed Stars AoE',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['662E', '6634'], source: 'Doomed Stars', capture: true }),
+      alertText: (data, matches, output) => {
+        const x = parseFloat(matches.x);
+        const y = parseFloat(matches.y);
+
+        data.storedStar = { x, y };
+
+        if (x < 100) {
+          if (y < 100)
+            return output.se?.();
+
+          return output.ne?.();
+        }
+        if (y < 100)
+          return output.sw?.();
+
+        return output.nw?.();
+      },
+      outputStrings: {
+        ne: Outputs.northeast,
+        nw: Outputs.northwest,
+        se: Outputs.southeast,
+        sw: Outputs.southwest,
+      },
+    },
+    {
+      id: 'Endsinger Elegeia',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['662C', '6682'], source: 'The Endsinger', capture: false }),
+      response: Responses.aoe(),
+    },
+    {
+      id: 'Endsinger Fatalism',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6632', source: 'The Endsinger', capture: false }),
+      condition: (data) => data.storedStar?.x !== undefined,
+      delaySeconds: 15,
+      alertText: (data, _matches, output) => {
+        const x = data.storedStar?.x;
+        const y = data.storedStar?.y;
+
+        delete data.storedStar;
+        if (!x || !y)
+          return undefined;
+
+        if (x < 100) {
+          if (y < 100)
+            return output.se?.();
+
+          return output.ne?.();
+        }
+        if (y < 100)
+          return output.sw?.();
+
+        return output.nw?.();
+      },
+      outputStrings: {
+        ne: Outputs.northeast,
+        nw: Outputs.northwest,
+        se: Outputs.southeast,
+        sw: Outputs.southwest,
+      },
+    },
+    {
+      id: 'Endsinger Galaxias',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6C69', source: 'The Endsinger', capture: false }),
+      response: Responses.knockback(),
+    },
+    {
+      id: 'Endsinger Elenchos Middle',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6644', source: 'The Endsinger', capture: false }),
+      response: Responses.getOut(),
+    },
+    {
+      id: 'Endsinger Elenchos Outsides',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6642', source: 'The Endsinger', capture: false }),
+      response: Responses.getIn(),
+    },
+    {
+      id: 'Endsinger Death\'s Embrace',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6649', source: 'The Endsinger', capture: false }),
+      response: Responses.spread(),
+    },
+    {
+      id: 'Endsinger Death\'s Embrace Feathers',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6649', source: 'The Endsinger', capture: false }),
+      delaySeconds: 5.7,
```

Comment:
Is there a reason to write the trigger against `startsUsing` here rather than just against `ability`? From the One Run's worth of log I have, it seems this one is always 5.7 seconds.

(Alternatively, what about `delayseconds: (_, matches) => parseFloat(matches.duration) - 0,`? This makes it immediately clear to the reader how soon before the mechanic resolution the trigger activates, rather than "here's a random-looking delay"